src/app/pages/authentication/login/login.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { AuthService } from '../../../core/service/auth.service';
+import { Router } from '@angular/router';
+import { ToastService } from 'src/app/core/service/toast.service';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css'],
+})
+export class LoginComponent implements OnInit {
+  loginForm: FormGroup;
+
+  constructor(
+    private authService: AuthService,
+    private router: Router,
+    private toastService: ToastService
+  ) {
+    this.loginForm = new FormGroup({
+      email: new FormControl('', [
+        Validators.required,
+        Validators.pattern(/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/),
+      ]),
+      password: new FormControl('', [Validators.required]),
+    });
+  }
+
+  ngOnInit(): void {}
+
+  goToRegistration() {
+    this.router.navigate(['/registration']);
+  }
+
+  onSubmit() {
+    this.authService.onLogin(this.loginForm.value).subscribe({
+      next: () => {
+        if (this.authService.getUserType() === 'Admin') {
+          this.router.navigate(['/dashboard']);
+          return;
```

Comment:
what is this return for?

Reply:
@aky-wtag vai without the return it will navigate the user to the passenger component. Then If the user is admin the auth guard will redirect the user to the unauthorized page. So i have used early return.